Model/Reclaim.php

```diff
@@ -60,7 +88,99 @@ public function reclaim()
      */
     public function getLog()
     {
-        return TODO;
+        $log = file($this->_dir->getPath('log') . '/klaviyo.log');
+        if ($log != "") {
+            return $log;
+        } else {
+            return array (
+                'message' => 'Unable to retrieve log file'
+            );
+        }
+    }
+
+    /**
+     * Cleans the Klaviyo log file
+     * 
+     * @api
+     * @param string $date
+     * @return boolean
+     */
+    public function cleanLog($date)
+    {
+        //get log file path and do the old switcheroo in preparation for cleaning
+        $path = $this->_dir->getPath('log') . '/klaviyo.log';
+        $old = $path . ".old";
+        rename($path, $old);
+
+        //open file streams
+        $input = fopen($old, "rb");
```

Comment:
only bad thing about files in php is it takes up a heap of memory.  is there a more efficient way to do this? we should thing about how actively we need to clean this / how to prevent us from reading a file that could be too big

Reply:
is this still a concern if we are using a filestream? my understanding is that this would only be loading one line of the file into memory at a time.

could be an issue in getLog fwiw, since that method does just load the entire contents of the file into memory. unsure how else we would be able to return a response without doing that though